### PR TITLE
Local provider Fix

### PR DESCRIPTION
**Pull request by ajinkya-dhamnaskar**
1. Added SECURITY_PROTOCOL column to LOCAL_SUBMISSION table.
2. Made security protocol mandatory for local job submission
3. Made local job submission to work with existing implementation.
4. Relaxed process folder privileges to 744 (earlier 644)
5. Provisioned local and third party file transfer in output staging for the same.

### CODE REVIEW

File: modules/registry/registry-core/src/main/resources/appcatalog-mysql.sql [hunk 1]
```diff
@@ -442,6 +442,7 @@ CREATE TABLE LOCAL_SUBMISSION
 (
         RESOURCE_JOB_MANAGER_ID VARCHAR (255) NOT NULL,
         JOB_SUBMISSION_INTERFACE_ID VARCHAR (255) NOT NULL,
+        SECURITY_PROTOCOL VARCHAR (255) NOT NULL,
```
Comment (lahirug) [inline modules/registry/registry-core/src/main/resources/appcatalog-mysql.sql]: How this is going to affect the production deployment ? Do we have a data migration mechanism in airavata ?

File: modules/gfac/gfac-impl/src/main/java/org/apache/airavata/gfac/impl/task/LocalJobSubmissionTask.java [hunk 2]
```diff
@@ -56,78 +61,85 @@ public void init(Map<String, String> propertyMap) throws TaskException {
...
-            standardOutWriter.join();
-            standardErrorWriter.join();
-
-            *//*
-             * check return value. usually not very helpful to draw conclusions based on return values so don't bother.
-             * just provide warning in the log messages
-             *//*
-            if (returnValue != 0) {
-                log.error("Process finished with non zero return value. Process may have failed");
+
```
Comment (lahirug) [inline modules/gfac/gfac-impl/src/main/java/org/apache/airavata/gfac/impl/task/LocalJobSubmissionTask.java]: Do you think this has to split in to few methods ? Looks bit long for a single method.

File: modules/orchestrator/orchestrator-core/src/main/java/org/apache/airavata/orchestrator/cpi/impl/SimpleOrchestratorImpl.java [hunk 3]
```diff
@@ -406,11 +406,17 @@ private String getTaskDag(List<String> taskIdList) {
...
-                        processOutput.setValue(appName + ".stdout");
+                        if(null == processOutput.getValue() ||
+                                (null != processOutput.getValue() && processOutput.getValue().trim().isEmpty())){
+                            processOutput.setValue(appName + ".stdout");
+                        }
                         createOutputDataSatagingTasks(processModel, gatewayId, dataStagingTaskIds, processOutput);
                         break;
                     case STDERR:
-                        processOutput.setValue(appName + ".stderr");
+                        if(null == processOutput.getValue() ||
```
Comment (shamrath) [inline modules/orchestrator/orchestrator-core/src/main/java/org/apache/airavata/orchestrator/cpi/impl/SimpleOrchestratorImpl.java]: Simplify the logic

File: modules/orchestrator/orchestrator-core/src/main/java/org/apache/airavata/orchestrator/cpi/impl/SimpleOrchestratorImpl.java [hunk 4]
```diff
@@ -406,11 +406,17 @@ private String getTaskDag(List<String> taskIdList) {
                 DataType type = processOutput.getType();
                 switch (type) {
                     case STDOUT :
-                        processOutput.setValue(appName + ".stdout");
+                        if(null == processOutput.getValue() ||
+                                (null != processOutput.getValue() && processOutput.getValue().trim().isEmpty())){
```
Comment (shamrath) [inline modules/orchestrator/orchestrator-core/src/main/java/org/apache/airavata/orchestrator/cpi/impl/SimpleOrchestratorImpl.java]: You can simplify this logic to (processOutput.getValue == null || processOutput.getvalue().trim().isEmpty())

**Pull request merged**